File: setima/c/5.c
#include <stdio.h>
#include <string.h>

struct Registro {
    char nome[50];
    char endereco[100];
    char telefone[20];
};

int main() {
    struct Registro agenda[10];
    int num_registros = 0;

    while (1) {
        int opcao;
        printf("Menu de Opções:\n");
        printf("1 - Cadastrar 10 registros\n");
        printf("2 - Pesquisar registro pelo nome\n");
        printf("3 - Classificar registros por ordem de nome\n");
        printf("4 - Apresentar todos os registros\n");
        printf("5 - Sair do programa\n");
        printf("Escolha uma opção: ");
        scanf("%d", &opcao);

        if (opcao == 1) {
            for (int i = num_registros; i < num_registros + 10; i++) {
                printf("Digite o nome: ");
                scanf("%s", agenda[i].nome);
                printf("Digite o endereço: ");
                scanf("%s", agenda[i].endereco);
                printf("Digite o telefone: ");
                scanf("%s", agenda[i].telefone);
            }
            num_registros += 10;
        } else if (opcao == 2) {
            char nome_pesquisa[50];
            printf("Digite o nome a ser pesquisado: ");
            scanf("%s", nome_pesquisa);
            int encontrado = 0;
            for (int i = 0; i < num_registros; i++) {
                if (strcmp(agenda[i].nome, nome_pesquisa) == 0) {
                    printf("Registro encontrado:\n");
                    printf("Nome: %s\n", agenda[i].nome);
                    printf("Endereço: %s\n", agenda[i].endereco);
                    printf("Telefone: %s\n", agenda[i].telefone);
                    encontrado = 1;
                    break;
                }
            }
            if (!encontrado) {
                printf("Registro não encontrado.\n");
            }
        } else if (opcao == 3) {
            // Supondo que você queira classificar por ordem alfabética de nomes
            for (int i = 0; i < num_registros; i++) {
                for (int j = i + 1; j < num_registros; j++) {
                    if (strcmp(agenda[i].nome, agenda[j].nome) > 0) {
                        struct Registro temp = agenda[i];
                        agenda[i] = agenda[j];
                        agenda[j] = temp;
                    }
                }
            }
        } else if (opcao == 4

File: setima/python/5.py
agenda = []

while True:
    print("Menu de Opções:")
    print("1 - Cadastrar 10 registros")
    print("2 - Pesquisar registro pelo nome")
    print("3 - Classificar registros por ordem de nome")
    print("4 - Apresentar todos os registros")
    print("5 - Sair do programa")
    
    opcao = input("Escolha uma opção: ")
    
    if opcao == "1":
        for _ in range(10):
            nome = input("Digite o nome: ")
            endereco = input("Digite o endereço: ")
            telefone = input("Digite o telefone: ")
            registro = (nome, endereco, telefone)
            agenda.append(registro)
    elif opcao == "2":
        nome_pesquisa = input("Digite o nome a ser pesquisado: ")
        for registro in agenda:
            if registro[0] == nome_pesquisa:
                print("Registro encontrado:")
                print(registro)
                break
        else:
            print("Registro não encontrado.")
    elif opcao == "3":
        agenda.sort(key=lambda x: x[0])
    elif opcao == "4":
        for registro in agenda:
            print("Nome:", registro[0])
            print("Endereço:", registro[1])
            print("Telefone:", registro[2])
            print()
    elif opcao == "5":
        break
    else:
        print("Opção inválida. Tente novamente.")
